jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererService.kt

```diff
@@ -322,6 +322,8 @@ class CodeWhispererService {
             }
         } else if (response.recommendations().isNotEmpty()) {
             updateCodeWhisperer(nextStates, isPopupShowing)
+            val perceivedLatency = CodeWhispererInvocationStatus.getInstance().getTimeSinceDocumentChanged()
+            CodeWhispererTelemetryService.getInstance().sendPerceivedLatencyEvent(requestId, requestContext, responseContext, perceivedLatency)
```

Comment:
Since this is a per-trigger event -- let's put this in `CodeWhispererPopupManager.showPopup()` https://github.com/zixlin7/aws-toolkit-jetbrains/blob/111d4986ee590516c4e7fb9e21d2310a71c34c2f/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/popup/CodeWhispererPopupManager.kt#L296
This is the path that's only reached when it's called first time and we are about to show the popup. Doing this will imply that this event will only be sent in those "shown" cases(Accepted or Rejected), which makes sense since for recommendations that're not shown, the concept of "perceived latency" does not apply.

And we want to make sure for VSC this is also true.

Reply:
@andrewyuq updated and it's already the case for VSC. Thanks!